ThirdPartyAudit failure for JDK19

### CI Link

https://gradle-enterprise.elastic.co/s/mq6p6fx7hwtoc

### Repro line

./gradlew :libs:elasticsearch-plugin-analysis-api:thirdPartyAudit -Druntime.java=19

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

Forbidden APIs output:
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfByte' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfByte'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfShort' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfShort'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfInt' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfInt'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfLong' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfLong'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfFloat' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfFloat'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySegment' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySegment'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.MemorySession' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.MemorySession'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfShort' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfShort'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfShort' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfShort'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfInt' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfInt'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfInt' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfInt'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfLong' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfLong'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfLong' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfLong'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfFloat' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfFloat'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfFloat' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfFloat'). Please fix the classpath!
==end of forbidden APIs==
Missing classes:
  * java.lang.foreign.MemorySegment
  * java.lang.foreign.MemorySession
  * java.lang.foreign.ValueLayout
  * java.lang.foreign.ValueLayout$OfByte
  * java.lang.foreign.ValueLayout$OfFloat
  * java.lang.foreign.ValueLayout$OfInt
  * java.lang.foreign.ValueLayout$OfLong
  * java.lang.foreign.ValueLayout$OfShort

just adding a `ignoreMissingClasses` section won't help here. It is not a problem in JDK<19 
and also in JDK19 when added that gradle configuration it does not fix the problem either
```
tasks.named("thirdPartyAudit").configure {
  ignoreMissingClasses(
    'java/lang/foreign/MemorySegment',
    'java/lang/foreign/MemorySession',
    'java/lang/foreign/ValueLayout',
    'java/lang/foreign/ValueLayout$OfByte',
    'java/lang/foreign/ValueLayout$OfFloat',
    'java/lang/foreign/ValueLayout$OfInt',
    'java/lang/foreign/ValueLayout$OfLong',
    'java/lang/foreign/ValueLayout$OfShort'
  )
}

```

in JDK19 
```
the same as previous beginning...
WARNING: Class 'java.lang.foreign.ValueLayout' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfFloat' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfFloat'). Please fix the classpath!
WARNING: Class 'java.lang.foreign.ValueLayout$OfFloat' cannot be loaded (while looking up details about referenced class 'java.lang.foreign.ValueLayout$OfFloat'). Please fix the classpath!
==end of forbidden APIs==

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':libs:elasticsearch-plugin-analysis-api:thirdPartyAudit'.
> All excluded classes seem to have no issues. This is sometimes an indication that the check silently failed
```

in JDK18
```
* What went wrong:
Execution failed for task ':libs:elasticsearch-plugin-analysis-api:thirdPartyAudit'.
> All excluded classes seem to have no issues. This is sometimes an indication that the check silently failed
```

I wonder why would it fail in `plugin-analysis-api` in the first place? It probably affects any other gradle module?

--- a/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ThirdPartyAuditPrecommitPlugin.java
+++ b/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ThirdPartyAuditPrecommitPlugin.java
@@ -16,7 +16,6 @@ import org.gradle.api.Project;
 import org.gradle.api.Task;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.tasks.TaskProvider;
-import org.gradle.internal.jvm.Jvm;
 
 import java.nio.file.Path;
 

--- a/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ThirdPartyAuditPrecommitPlugin.java
+++ b/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ThirdPartyAuditPrecommitPlugin.java
@@ -61,7 +61,9 @@ public class ThirdPartyAuditPrecommitPlugin extends PrecommitPlugin {
                 return dep.getGroup() != null && dep.getGroup().startsWith("org.elasticsearch") == false;
             }));
             t.dependsOn(resourcesTask);
-            t.setJavaHome(Jvm.current().getJavaHome().getPath());
+            if (BuildParams.getIsRuntimeJavaHomeSet()) {
+                t.setJavaHome(BuildParams.getRuntimeJavaHome().getPath());
+            }
             t.getTargetCompatibility().set(project.provider(BuildParams::getRuntimeJavaVersion));
             t.setSignatureFile(resourcesDir.resolve("forbidden/third-party-audit.txt").toFile());
             t.getJdkJarHellClasspath().from(jdkJarHellConfig);

